ui/src/components/SampleControls/SnapshotControl.jsx

```diff
@@ -25,19 +25,30 @@ function SnapshotControl(props) {
     return sourceScale * imageRatio;
   });
 
-  const takeSnapshot = useCallback(() => {
+  const takeSnapshot = useCallback(async () => {
     const img = document.querySelector('#sample-img');
     const fimg = new fabric.Image(img);
     fimg.scale(imageRatio);
 
     canvas.setBackgroundImage(fimg);
     canvas.renderAll();
 
-    const filename = `${proposal}-${currentSampleName}.jpeg`;
-    download(filename, canvas.toDataURL({ format: 'jpeg' }));
+    const imgDataURI = canvas
+      .toDataURL({
+        format: 'png',
+        backgroundColor: null,
+      })
+      .split(',')[1];
```

Comment:
Could this split be done in the back-end? It would make the API contract simpler: a full data URI instead of just the image payload. This way the back-end could also check that it's receiving the expected PNG mime type.

Reply:
That's a great suggestion, I will make the necessary changes